packages/suite-desktop/next.config.js

```diff
@@ -20,13 +21,11 @@ const gitRevisionPlugin = new GitRevisionPlugin();
 
 // TODO: After Raf's tschuss-next is merged, move keys to some constants file to avoid duplication in suite-web and suite-desktop package
 // TODO: Do not forget to move process.env.PUBLIC_KEY and process.env.STABLE_CONFIG to new webpack files
-const productionJwsPublicKey =
-    process.env.JWS_PUBLIC_KEY &&
-    `-----BEGIN PUBLIC KEY-----
-${process.env.JWS_PUBLIC_KEY}
------END PUBLIC KEY-----`;
 
-const developmentJwsPublicKey = `-----BEGIN PUBLIC KEY-----
+// use key from JWS_PUBLIC_KEY_FILE or use the default devel public key
+const JWS_PUBLIC_KEY = process.env.JWS_PUBLIC_KEY_FILE
+    ? fs.readFileSync(process.env.JWS_PUBLIC_KEY_FILE, 'utf-8')
+    : `-----BEGIN PUBLIC KEY-----
```

Comment:
I'm not sure in what context these key loading snippets are run, but I'd expect they have access to whether they're running in a production or development environment. In such case I'd prefer to not fallback to the devel key by default but fail loudly. That would prevent using devel keys in production. Something along the lines of

```javascript
if (isProduction) {
  const JWS_PUBLIC_KEY = fs.readFileSync(process.env.JWS_PUBLIC_KEY_FILE, 'utf-8') || throw Error('key could not be loaded')
} else {
  const JWS_PUBLIC_KEY = `<The devel key>'
}
```

Reply:
@tomasklim Could you maybe eleborate on what is the best way to obtain the `isProduction` variable in these contexts (if that's even possible).

Reply:
Good catch @goodhoko! 

I'd also prefer not setting this via

```
'process.env.STABLE_CONFIG': !!process.env.JWS_PUBLIC_KEY_FILE,
```

but rather

```
'process.env.STABLE_CONFIG': isProduction,
```

Reply:
The distinction that needs to be made here isn't between production/development build but codesign[1]/every other build.

The production private keys are provided to the codesign jobs via this mechanism:
- codesign jobs only run on `codesign` branch
- `codesign` branch is protected (as the only one)
- private keys are set as [protected environment variables](https://docs.gitlab.com/ee/ci/variables/README.html#project-cicd-variables) in GitLab GUI and hence are only available to jobs running on protected branches.

We discussed this with @tomasklim and propose this solution:

We'll levarage the current setup and use the `CI_COMMIT_REF_PROTECTED` [predefined CI variable](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html) which is `true` for jobs thar run on protected branches to identify jobs that should use the production keys. Loading the key safely should then be as easy as

```js
const isCodesign = process.env.CI_COMMIT_REF_PROTECTED === 'true';
const key = isCodesign ? loadProductionKeyOrFailLoudly() : "<development key>"
```

This should prevent accidentally fallbacking in codesign jobs as could happen when one uses the presence of given variable itself as the `isCodesign` flag.

Reply:
edit: I deleted my last comment as it might disclose possibly sensitive information. Will contact you both at Slack.